plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -215,12 +227,24 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
             throws SQLException
     {
         PreparedStatement statement = connection.prepareStatement(sql);
-        if (statement.isWrapperFor(Statement.class)) {
-            statement.unwrap(Statement.class).enableStreamingResults();
+        if (statement.isWrapperFor(JdbcStatement.class)) {
+            statement.unwrap(JdbcStatement.class).enableStreamingResults();
         }
         return statement;
     }
 
+    private static int getTimestampPrecision(int timestampColumnSize)
+    {
+        if (timestampColumnSize == ZERO_PRECISION_TIMESTAMP_COLUMN_SIZE) {
+            return 0;
+        }
+        else {
+            int timestampPrecision = timestampColumnSize - ZERO_PRECISION_TIMESTAMP_COLUMN_SIZE - 1;
+            verify(timestampPrecision <= MAX_SUPPORTED_TIMESTAMP_PRECISION);
```

Comment:
in case this ever fails, quote actual value of `timestampPrecision`

also, validate the value is within range (>=1)

```suggestion
            verify(1 <= timestampPrecision && timestampPrecision <= MAX_SUPPORTED_TIMESTAMP_PRECISION, "Unexpected timestamp precision %s calculated from timestamp column size %s", timestampPrecision, timestampColumnSize);
```